File: programas_java/IOT/BancoAD.py

class BancoAD:

    def capturar(self, datos):
        
        # 1. Abrir el archivo Clientes.txt
        archivoOut = open("Clientes.txt", "a")
        
        # 2. Almacenar los datos en el archivo
        archivoOut.write(datos+"\n")
        
        # 3. Cerrar el archivo
        archivoOut.close()
        
        return "Datos capturados: "+datos

    def consultar(self):
        # 1. Abrir el archivo para leer
        archivoIn = open("Clientes.txt","r")
        
        # 2. Leer los datos del archivo
        datos=""
        cliente = archivoIn.readline()
        while cliente != "":
            #datos = datos + cliente + "\n"
            datos = datos + cliente
            cliente = archivoIn.readline()
        
        # 3. Cerrar el archivo
        archivoIn.close()
        
        return datos

    def consultarTipo(self,tipo):
        # 1. Abrir el archivo
        archivoIn = open("Clientes.txt","r")

        # 2. Procesar datos del archivo
        datos = ""
        encontrado = False
        cta=""
        nom=""
        tcta=""
        saldo=""
        
        cliente = archivoIn.readline()
        while cliente != "":
            st    = cliente.split("_")
            cta   = st[0]
            nom   = st[1]
            tcta  = st[2]
            saldo = st[3]
            
            if tcta == tipo:
                datos = datos + cliente
                encontrado = True
        
            cliente = archivoIn.readline()
        
        # 3. Cerrar el archivo
        archivoIn.close
        
        #if encontrado == False:
        if (not encontrado):
            datos = "No se localizo el Tipo de cuenta "+tipo

        return datos

    def consultarCta(self,ncta):
        # 1. Abrir el archivo
        archivoIn = open("Clientes.txt","r")
        
        # 2. Procesar datos del archivo
        datos = ""
        encontrado = False
        cta=""
        nom=""
        tcta=""
        saldo=""
        
        cliente = archivoIn.readline()
        while (cliente != "" and not encontrado):
            st = cliente.split("_")
            cta = st[0]
            nom = st[1]
            tcta= st[2]
            saldo = st[3]
            
            if cta == ncta:
                datos = datos + cliente
                encontrado = True
            
            cliente = archivoIn.readline()
        
        # 3. Cerrar el archivo
        archivoIn.close
        
        #if encontrado == False:
        if (not encontrado):
            datos = "No se localizo el Cliente con Cuenta "+ncta
        
        return datos

File: programas_java/ConstrucciondeSoftware/DB-JDBC/BancoAD.java
import java.io.PrintWriter;
import java.io.FileWriter;
import java.io.BufferedReader;
import java.io.FileReader;;
import java.io.IOException;
import java.io.FileNotFoundException;

import java.util.StringTokenizer;

public class BancoAD
{
	private PrintWriter archivoOut;
	private BufferedReader archivoIn;

	public String capturar(String datos)
	{
		String resultado = "";

		try{
			// Abrir el archivo para almacenar los datos
			archivoOut = new PrintWriter(new FileWriter("Clientes.txt", true));

			// Guardar o almaccenar los datos
			archivoOut.println(datos);

			// Cerrar el archivo
			archivoOut.close();

			resultado = "Datos capturados...";
		}
		catch(IOException ioe){
			resultado = "Error con el archivo";
			System.out.println("Error: " + ioe);
		}
		
		return resultado;
	}

	public String consultar(){
		String datos = "";
		String cliente;

		try{
			// Abrir el archivo
			archivoIn = new BufferedReader(new FileReader("Clientes.txt"));

			// Procesar el archivo para obtener los datos
			while(archivoIn.ready()){
				cliente = archivoIn.readLine();
				
				System.out.println(cliente);
				
				datos += cliente + "\n";
			}

			// Cerrar el archivo
			archivoIn.close();
		}
		catch(FileNotFoundException fnfe){
			datos = "Error al abrir el archivo";
			System.out.println("Error: " + fnfe);
		}
		catch(IOException ioe){
			datos = "Error al cerrar el archivo";
			System.out.println("Error: " + ioe);
		}
		return datos;
	}

	public String consultarTipo(String tcta){
		String datos = "";
		String cliente;
		StringTokenizer stCliente;
		String ncta, nombre, tipo, saldo;
		boolean encontrado = false;

		try{
			// Abrir el archivo
			archivoIn = new BufferedReader(new FileReader("Clientes.txt"));

			// Procesar el archivo
			while(archivoIn.ready()){
				cliente = archivoIn.readLine();

				stCliente = new StringTokenizer(cliente, "_");
				ncta = stCliente.nextToken();
				nombre = stCliente.nextToken();
				tipo = stCliente.nextToken();
				saldo = stCliente.nextToken();

				if(tcta.equals(tipo)){
					datos += cliente + "\n";
					encontrado = true;
				}
			}
			
			// Cerrar el archivo
			archivoIn.close();
			if(!encontrado)
				datos = "No se localizó el tipo " + tcta;
		}
		catch(FileNotFoundException fnfe){
			datos = "Error al abrir el archivo";
			System.out.println("Error: " + fnfe);
		}
		catch(IOException ioe){
			datos = "Error al cerrar el archivo";
			System.out.println("Error: " + ioe);
		}
		return datos;
	}

	public String consultarNcta(String ncta){
		String datos = "";
		String cliente;
		StringTokenizer stCliente;
		String cuenta, nombre, tipo, saldo;
		boolean encontrado = false;

		try{
			// Abrir el archivo
			archivoIn = new BufferedReader(new FileReader("Clientes.txt"));

			// Procesar el archivo
			while(archivoIn.ready()){
				cliente = archivoIn.readLine();

				stCliente = new StringTokenizer(cliente, "_");
				cuenta = stCliente.nextToken();
				nombre = stCliente.nextToken();
				tipo = stCliente.nextToken();
				saldo = stCliente.nextToken();

				if(cuenta.equals(ncta)){
					datos += cliente + "\n";
					encontrado = true;
				}
			}
			
			// Cerrar el archivo
			archivoIn.close();
			if(!encontrado)
				datos = "No se localizó el tipo " + ncta;
		}
		catch(FileNotFoundException fnfe){
			datos = "Error al abrir el archivo";
			System.out.println("Error: " + fnfe);
		}
		catch(IOException ioe){
			datos = "Error al cerrar el archivo";
			System.out.println("Error: " + ioe);
		}
		return datos;
	}
}